packages/algoliasearch-helper/src/algoliasearch.helper.js

```diff
@@ -1631,10 +1631,16 @@ AlgoliaSearchHelper.prototype._recommend = function () {
     return derivedState._buildQueries(derivedIndex);
   });
 
-  var queries = Array.prototype.concat.apply(
-    this.recommendState._buildQueries(index),
-    derivedQueries
-  );
+  var queries = Array.prototype.concat
+    .apply(this.recommendState._buildQueries(index), derivedQueries)
+    .map(function (query) {
+      return Object.assign(query, {
+        queryParameters: Object.assign(query.queryParameters || {}, {
+          clickAnalytics: searchState.clickAnalytics,
+          userToken: searchState.userToken,
```

Comment:
Q: If the client passes `queryParameters` to the recommend widget, do they "lose" automatic event collection ?

Reply:
This is augmented thanks to `Object.assign()`, but it's a good idea to have a specific test for that, I'll add it to the PR.